consul/client.go

```diff
@@ -12,14 +12,51 @@ import (
 	"time"
 
 	"github.com/hashicorp/consul/consul/structs"
+	"github.com/hashicorp/consul/lib"
 	"github.com/hashicorp/serf/coordinate"
 	"github.com/hashicorp/serf/serf"
 )
 
 const (
```

Comment:
If these were put into the config structure, even if they aren't exposed to the user, you could cook up a unit test to run this through the different modes by setting smaller intervals and rates. This would be good because there's a fair amount of logic here, and it's a pretty crucial part of the code.